### PR TITLE
2267 add service reporting

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Fixes service display in UI #2267

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running manually
* [x] If applicable, add screenshots or log transcripts of the feature working
image

### CODE REVIEW

File: monkey/monkey_island/cc/resources/node.py [hunk 1]
```diff
@@ -8,10 +10,26 @@
 class Node(AbstractResource):
     urls = ["/api/netmap/node"]
 
+    def __init__(self, node_repository: INodeRepository):
+        self.node_repository = node_repository
+
     @jwt_required
     def get(self):
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/resources/node.py:17]: This could use a test, but make sure it's the right approach in the first place

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Overall, I'm not a fan of this approach. It's another example of the frontend driving the backend, resulting in increased coupling. `get_receiver_nodes()` has been added to `INodeRepository` because the frontend wants the data in a certain format. If the data format that the frontend wants changes, we'll need to change the repository to match. This is unnecessary coupling.

Additionally, `/api/netmap/nodes` now returns two different data structures, which is API sacrilege.

Let's just have the frontend transform the data into the format it needs. In other words, leave the presentation of the data to the presentation layer.

**Comment by VakarisZ** [comment]
> `get_receiver_nodes()` has been added to `INodeRepository` because the frontend wants the data in a certain format.

Formatting and filtering are two different things. `get_receiver_nodes()` does not change the format of `Node` model. Though it should return Nodes, not just ID's.

To get scanned machines you would currently need to do 2 joins: first get all nodes and join target nodes if scanned is in the communications. Then you would need to join machines based on those nodes. If you need to download 2 full tables from a database for a subset of one it means that API is not designed for your use case.
We need to determine if the API should answer the question "which machines got scanned/exploited". It would be a bit hypocritical to say no when our own front-end needs this question answered.


> Let's just have the frontend transform the data into the format it needs. In other words, leave the presentation of the data to the presentation layer.

We should also leave database operations to the database. Filters and joins are most effectively done by the database

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ Changelog](https://keepachangelog.com/en/1.0.0/).
 - `/api/agent-signals` endpoint. #2261
 - `/api/agent-logs/<uuid:agent_id>` endpoint. #2274
 - `/api/machines` endpoint. #2362
+- status parameter to `/api/node` endpoint. #2362
```
Comment (mssalvatore) [inline CHANGELOG.md:30]:
```suggestion
```

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**